### PR TITLE
Add OS-Specific Dialogs for Loading Files and Folders

### PR DESCRIPTION
**Pull request by MilesLitteral**
This fork adds Monomer.Common.OSDialogs, a module which allows users to quickly and easily retrieve file paths from their computer ,regardless of OS, and send the contents back into Monomer painlessly. The windows take configs of their own if necessary and can be used like this: 

```haskell
handleEvent ... = 
...
PromptFileDialog -> [ Task $ do
     sr <- osSelectFileDialog "Select Intake JSON" "./" ["*.json"] "json files" False
     case sr of 
       Nothing  -> return InvalidFile -- Failure Event
       Just val -> return $ ProcessFile val
]
ProcessFile file -> [Model $ model & filepath .~ (head file)]
```

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -45,6 +45,7 @@ packages:
 # (e.g., acme-missiles-0.3)
 extra-deps:
   - nanovg-0.8.0.0@sha256:0183b4295ccc2dfb94a7eca977fb45759e1480652578936aa7b71bb4b9626480,4742
+  - tinyfiledialogs-0.2.1.0@sha256:118f9fd7b721de68f1e84c8b820784e31140bf34e32f23fd6410fb57914a309c,1905
```
Comment (TrueWarg) [inline stack.yaml:48]: I think that comments should be have more specific name. Instead "Update <somethinf>" "Add tinyfiledialogs to deps".

**Review by TrueWarg — COMMENTED** [review]
It seems to me that the promptFileDialog here looks like something foreign, an extra addiction. This could just be added as an example, but not as part of the monomer, imho.

File: package.yaml [hunk 2]
```diff
@@ -51,6 +51,7 @@ dependencies:
   - transformers >= 0.5 && < 0.7
   - vector >= 0.12 && < 0.14
   - wreq >= 0.5.2 && < 0.6
+  - tinyfiledialogs
```
[the remark this answers is not in the document]
Comment (MilesLitteral) [reply package.yaml:54]: Yes, this is a case of the library being very stable across multiple operating systems, though a specific version can be specified if preferred

File: Common/BasicTypes.hs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+License     : BSD-3-Clause (see the LICENSE file)
+Maintainer  : fjvallarino@gmail.com
+Stability   : experimental
+Portability : non-portable
+
+Basic types used across the library.
+-}
+{-# LANGUAGE DeriveGeneric #-}
+
+module Monomer.Common.BasicTypes where
```
[the remark this answers is not in the document]
Comment (MilesLitteral) [reply Common/BasicTypes.hs:13]: Monomer.Common.BasicTypes, I have not made changes to; it should be equal in content to fjvallarino/monomer (main/src/common) 

Monomer.Common.OSDialogs is an entirely new module

**Comment by fjvallarino** [comment]
@MilesLitteral sorry for the long delay, I've been dealing with personal issues, and I'm slowly resuming work on the library.

I think this would be great to have as part of a recipes section I will be adding in the next few weeks, but I don't think it belongs as part of the library itself, since it adds an external dependency that not all users would be interested in.

I've been reducing the number of dependencies, and I avoid non-Haskell ones since they can become a maintenance problem given the multi-platform nature of the library.

**Comment by MilesLitteral** [comment]
I understand as, at best, it adds some native "flavoring" to monomer style-wise without too much extra utility (apart from optional convenience), looking forward to all future updates!

**Pull request closed without merging**